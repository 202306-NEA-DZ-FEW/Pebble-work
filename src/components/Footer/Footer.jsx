import React from "react";
import Link from "next/link";
import Translat from "@/util/Translat";

const Footer = (props) => {
    const { t } = Translat(props);
    return (
        <footer
            data-testid='footer'
            className=' md:flex md:items-center md:justify-center flex items-center justify-center bg-[#FDA855]'
            style={{
                width: "100%",
                height: "180px",
                zIndex: "90",
            }}
        >
            <div className='md:flex md:flex-row md:justify-between flex flex-col items-center md:xl:gap-72 md:gap-20 md:items-center'>
                <div>
                    <Link href='/' className='flex items-center'>
                        <img
                            src='/logo/Logo.png'
                            className='md:h-20 md:mr-3 h-10'
                            alt='Logo'
                        />
                    </Link>
                </div>
                <div className='flex flex-col items-center md:flex md:flex-col md:items-center md:gap-5'>
                    <ul
                        style={{
                            color: "black",
                            fontSize: "20px",
                            fontFamily: "Rubik",
                            fontWeight: "700",
                            lineHeight: "30px",
                            letterSpacing: "0.10px",
                            wordWrap: "break-word",
                        }}
                        className='flex gap-[30px] '
                    >
                        <li>
                            <Link href='/events'>Events</Link>
                        </li>
                        <li>
                            <Link href='/about'>About</Link>
                        </li>
                        <li>
                            <Link href='#'>How it works</Link>
                        </li>
                    </ul>
                    <div>
                        <p
                            style={{
                                color: "#878787",
                                fontSize: "17px",
                                fontFamily: "Rubik",
                                fontWeight: "300",
                                lineHeight: "30px",
                                letterSpacing: "0.10px",
                                wordWrap: "break-word",
                            }}
                        >
                            Â© Pebble Work, 2023
                        </p>
                    </div>
                </div>
                <div className='md:flex '>
                    <ul className='flex gap-20 md:gap-8'>
                        <li>
                            <Link href='#'>
                                <img src='/icons/Facebook.png' alt='Facebook' />
                            </Link>
                        </li>
                        <li>
                            <Link href='#'>
                                <img src='/icons/Google.png' alt='Gmail' />
                            </Link>
                        </li>
                        <li>
                            <Link href='#'>
                                <img src='/icons/Twitter.png' alt='Twitter' />
                            </Link>
                        </li>
                    </ul>
                </div>
            </div>
        </footer>
    );
};

export default Footer;
