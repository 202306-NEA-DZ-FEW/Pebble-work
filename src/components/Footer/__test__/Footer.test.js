import { render } from "@testing-library/react";
import React from "react";

import Footer from "../Footer"; // adjust this import to your file structure

describe("Footer", () => {
    it("renders correctly", () => {
        const { asFragment } = render(<Footer />);
        expect(asFragment()).toMatchSnapshot();
    });
});
