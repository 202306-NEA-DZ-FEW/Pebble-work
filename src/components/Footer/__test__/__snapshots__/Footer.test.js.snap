// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<DocumentFragment>
  <footer
    class="md:flex md:items-center md:justify-center flex items-center justify-center bg-[#FDA855]"
    data-testid="footer"
    style="width: 100%; height: 180px; z-index: 90;"
  >
    <div
      class="md:flex md:flex-row md:justify-between flex flex-col items-center md:xl:gap-72 md:gap-20 md:items-center"
    >
      <div>
        <a
          href="/"
        >
          <img
            alt="Logo"
            class="md:h-20 md:mr-3 h-10"
            src="/logo/Logo.png"
          />
        </a>
      </div>
      <div
        class="flex flex-col items-center md:flex md:flex-col md:items-center md:gap-5"
      >
        <ul
          class="flex gap-[30px]"
        >
          <li>
            <a
              href="/events"
            >
              footer.events
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              footer.about
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              footer.howItWorks
            </a>
          </li>
        </ul>
        <div>
          <p
            class="text-[#878787] text-17px"
          >
            Â© Pebble Work, 2023
          </p>
        </div>
      </div>
      <div
        class="md:flex "
      >
        <ul
          class="flex gap-20 md:gap-8"
        >
          <li>
            <a
              href="#"
            >
              <img
                alt="Facebook"
                class="md:w-[50px] w-[30px]"
                src="/icons/Facebook.svg"
              />
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              <img
                alt="Gmail"
                class="md:w-[50px] w-[30px]"
                src="/icons/Google.svg"
              />
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              <img
                alt="Github"
                class="md:w-[50px] w-[30px]"
                src="/icons/Github.svg"
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
